tests/devices/test_default_qutrit.py

```diff
@@ -1391,3 +1396,553 @@ def test_density_matrix_all_wires(self, qutrit_device_2_wires, wires, expected):
+        for QNodes with real-valued outputs"""
+        import jax
+        from jax import numpy as jnp
+
+        jax.config.update("jax_enable_x64", enable_x64)
+        p = jnp.array(0.543)
+        dev = qml.device("default.qutrit", wires=3, r_dtype=r_dtype)
+
+        @qml.qnode(dev, interface="jax", diff_method="backprop")
+        def circuit(x):
+            qml.TRX(x, wires=0)
+            return qml.apply(measurement)
+
+        if use_jit:
+            circuit = jax.jit(circuit)
+
+        res = circuit(p)
+        assert res.dtype == r_dtype
+
+    @pytest.mark.parametrize("enable_x64, c_dtype", [(False, np.complex64), (True, np.complex128)])
+    def test_complex_dtype(self, enable_x64, c_dtype, use_jit):
+        """Test that the user-defined dtype of the device is preserved
+        for QNodes with complex-valued outputs"""
+        import jax
+        from jax import numpy as jnp
+
+        jax.config.update("jax_enable_x64", enable_x64)
+        p = jnp.array(0.543)
+        dev = qml.device("default.qutrit", wires=3, c_dtype=c_dtype)
+
+        @qml.qnode(dev, interface="jax", diff_method="backprop")
+        def circuit(x):
+            qml.TRX(x, wires=0)
+            return qml.state()
+
+        if use_jit:
+            circuit = jax.jit(circuit)
+
+        res = circuit(p)
+        assert res.dtype == c_dtype
+
+
+@pytest.mark.jax
+@pytest.mark.parametrize("use_jit", [True, False])
+class TestPassthruIntegrationJax:
+    """Tests for integration with the PassthruQNode"""
+
+    def test_backprop_gradient(self, tol, use_jit):
+        """Tests that the gradient of the qnode is correct"""
+        import jax
+        from jax import numpy as jnp
+
+        dev = qml.device("default.qutrit", wires=2)
+
+        @qml.qnode(dev, diff_method="backprop", interface="jax")
+        def circuit(a, b):
+            qml.TRX(a, wires=0)
+            qml.TAdd(wires=[0, 1])
+            qml.TRY(b, wires=1, subspace=[0, 2])
+            return qml.expval(qml.GellMann(0, 3) @ qml.GellMann(1, 3))
+
+        if use_jit:
+            circuit = jax.jit(circuit)
+
+        a = jnp.array(-0.234)
+        b = jnp.array(0.654)
+
+        res = circuit(a, b)
+        expected_cost = 0.25 * (jnp.cos(a) * jnp.cos(b) - jnp.cos(a) + jnp.cos(b) + 3)
+        assert jnp.allclose(res, expected_cost, atol=tol, rtol=0)
+        res = jax.grad(circuit, argnums=(0, 1))(a, b)
+        expected_grad = jnp.array(
+            [
+                -0.25 * (jnp.sin(a) * jnp.cos(b) - jnp.sin(a)),
+                -0.25 * (jnp.cos(a) * jnp.sin(b) + jnp.sin(b)),
+            ]
+        )
+
+        assert jnp.allclose(jnp.array(res), jnp.array(expected_grad), atol=tol, rtol=0)
+
+    def test_backprop_gradient_broadcasted(self, tol, use_jit):
+        """Tests that the gradient of the broadcasted qnode is correct"""
+        import jax
+        from jax import numpy as jnp
+
+        if use_jit:
+            pytest.skip()
```

Comment:
This one here is skipped because of the issue w/broadcasting right?

Reply:
yep!